make/prepare

```diff
@@ -308,8 +311,28 @@ if protocol == "https":
             ssl_cert = os.path.join("/etc/nginx/cert", os.path.basename(target_cert_path)),
             ssl_cert_key = os.path.join("/etc/nginx/cert", os.path.basename(target_cert_key_path)))
 else:
-    render(os.path.join(templates_dir, "nginx", "nginx.http.conf"),
-        nginx_conf)
+    render(os.path.join(templates_dir, "nginx", "nginx.http.conf"), nginx_conf)
+    custom_nginx_location_file_pattern = 'harbor.http.*.conf'
+
+def add_additional_location_config(src, dst):
+    """
+    This conf file is used for user that wanna add additional customized locations to harbor proxy
+    :params src: source of the file
+    :params dst: destination file path
+    """
+    if not os.path.isfile(src):
+        return
+    print("Copying nginx configuration file {src} to {dst}".format(src=src, dst=dst))
+    shutil.copy2(src, dst)
+    mark_file(dst)
+
+nginx_template_ext_dir = os.path.join(templates_dir, 'nginx', 'ext')
+if os.path.exists(nginx_template_ext_dir):
```

Comment:
We can remove `add_additional_location_config`, and copy the directory to target `conf.d` directory, and call mark_file to `conf.d`.